### PR TITLE
Improve `nvim-remote` mode

### PR DESCRIPTION
**Pull request by cmoog**
- If _not_ inside a neovim session, treat as a normal nvim invocation and suspend lazygit.

- If inside a neovim session:
  - Do not try to suspend lazygit.
  - Send `q` keystroke to neovim session to quit lazygit.
  - Send filename/line/etc. to neovim session.

- **Please check if the PR fulfills these requirements**

* [x] Cheatsheets are up-to-date (run `go generate ./...`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [ ] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [ ] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [ ] Docs (specifically `docs/Config.md`) have been updated if necessary
* [x] You've read through your own file changes for silly mistakes etc

### CODE REVIEW

File: pkg/config/editor_presets.go [hunk 1]
```diff
@@ -42,22 +44,27 @@ type editPreset struct {
 	editAtLineTemplate        string
 	editAtLineAndWaitTemplate string
 	openDirInEditorTemplate   string
-	suspend                   bool
+	suspend                   func() bool
 }
 
+func _const(a bool) func() bool { return (func() bool { return a }) }
```
Comment (jesseduffield) [inline pkg/config/editor_presets.go:50]: Can we rename '_const' to 'returnBool'

**Review by jesseduffield — CHANGES_REQUESTED** [review]
Code looks good, just one comment

**Comment by jesseduffield** [comment]
I've taken the liberty of squashing the commits into one. Once tests pass we can merge

**Comment by xzbdmw** [comment]
Are there easy ways to rebind quit in lazygit to a custom command in nvim such as hide terminal?

**Comment by dudicoco** [comment]
> Are there easy ways to rebind quit in lazygit to a custom command in nvim such as hide terminal?

@cmoog @jesseduffield can you please elaborate on why this PR changes the behavior and quits lazygit instead of suspending it? For me the previous behavior was perfect and this actually feels like a downgrade in experience.

**Comment by jesseduffield** [comment]
@dudicoco I am regrettably not a neovim daily user so I probably can't do justice to the motivation but my assumption was that if you want to edit random files from the files view, you'd rather switch to the parent neovim process than to create another child one, because by creating another child one, you can end up with many layers of neovim->lazygit->neovim->etc nesting.

Could you elaborate on what specifically you liked about the previous behaviour and what you don't like about the current behaviour?

**Comment by dudicoco** [comment]
@jesseduffield what you have described is the behavior of `nvim-remote` before this PR as well.
I'm referring specifically to the change in this PR which sends `q` keystroke to neovim session to quit lazygit instead of suspending it.

**Pull request merged**